SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -490,12 +490,32 @@ void ClearGeneratedSchema()
 	DeleteGeneratedSchemaFiles();
 }
 
-void TryLoadExistingSchemaDatabase()
+bool TryLoadExistingSchemaDatabase()
 {
-	const USchemaDatabase* const SchemaDatabase = Cast<USchemaDatabase>(FSoftObjectPath(TEXT("/Game/Spatial/SchemaDatabase.SchemaDatabase")).TryLoad());
+	const FString SchemaDatabasePackagePath = TEXT("/Game/Spatial/SchemaDatabase");
+	const FString SchemaDatabaseAssetPath = FString::Printf(TEXT("%s.SchemaDatabase"), *SchemaDatabasePackagePath);
+	const FString SchemaDatabaseFileName = FPackageName::LongPackageNameToFilename(SchemaDatabasePackagePath, FPackageName::GetAssetPackageExtension());
 
-	if (SchemaDatabase != nullptr)
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+
+		FFileStatData StatData = PlatformFile.GetStatData(*SchemaDatabaseFileName);
+
+	if (StatData.bIsValid)
 	{
+		if (StatData.bIsReadOnly)
+		{
+			UE_LOG(LogSpatialGDKSchemaGenerator, Error, TEXT("Schema Database at %s%s is not writable."), *SchemaDatabasePackagePath, *FPackageName::GetAssetPackageExtension());
```

Comment:
Can we make the error message say that they should make the file writable in order to successfully generate schema